### PR TITLE
Add stats CLI command to report rabies vaccines by month (#185)

### LINKED ISSUE #185: Create an entry point to call count how many rabies vaccines we applied by month
As **user**, I want to have an command and entry point to get how many rabies vaccines we applied by month
 **so that** I can easily call this method

**Acceptance Criteria**
- I can receive year argument from command line
- I can see a mehod named `genrate_stats` in the [main.py file](https://github.com/josdem/py-vetlog-buddy/blob/main/src/vetlog_buddy/main.py)
- I can see a project script added [here](https://github.com/josdem/py-vetlog-buddy/blob/main/pyproject.toml)
- I can execute a command with year argument as follow: `uv run stats --year ${year}`
  - Where `${year}` is the year we want to get  how many rabies vaccines we applied by month
- All test passing

### PR DESCRIPTION
**Pull request by satyam-311**
Closes #185

Adds a `stats` CLI command that shows how many rabies vaccines were applied per month for a given year. Run it like:

`uv run stats --year 2026`

Main changes:
- Added `generate_stats(year)` and `stats_cli()` in main.py, following the same pattern as the existing `vaccines` command
- Registered `stats` as a script entry in pyproject.toml
- Had to rename `VaccinationService`'s constructor param from `repository` to `vaccinationRepository` since that's what the test in test_vaccine_service.py expected — updated all the internal references and the one existing test that used the old name
- All 62 tests pass now, including the one that was failing before

One note: the issue spells it `genrate_stats` but I'm pretty sure that's a typo, so I went with the correct spelling `generate_stats`. Let me know if you'd rather I match it exactly.

### CODE REVIEW

**Comment by satyam-311** [comment]
@josdem this is ready for review whenever you get a chance. Let me know if anything needs adjusting.

File: src/vetlog_buddy/main.py [hunk 1]
```diff
@@ -102,6 +102,30 @@ def vaccinations_cli():
     vaccines(id=args.id)
 
 
+def generate_stats(year: int):
+    """Generate monthly rabies vaccination stats"""
```
Comment (josdem) [inline src/vetlog_buddy/main.py:106]: Please add test to cover `generate_stats` and `stats_cli`, you can find examples how to do it here: https://github.com/josdem/py-vetlog-buddy/blob/f862a77483e70cfd5432cbb05971171f865ab5cb/tests/unit/test_main.py#L68L118

**Review by josdem — CHANGES_REQUESTED** [review]
Just an small change before merge, good job @satyam-311 👍

**Comment by josdem** [comment]
@all-contributors please add @satyam-311 for code

File: src/vetlog_buddy/main.py (hunk 1 above)
Comment (satyam-311) [reply src/vetlog_buddy/main.py:106]: Added tests for both generate_stats and stats_cli, following the same pattern as test_vaccines/test_vaccination_cli. All 64 tests pass now.

**Review by josdem — APPROVED** [review]
Thanks for doing suggested changes, now it looks good to me 👍

**Review by Shouryaverma19 — COMMENTED** [review]
The architectural refactoring looks clean, and the unit tests cover the CLI parser edge cases beautifully. Ready from my side!

**Comment by josdem** [comment]
> The architectural refactoring looks clean, and the unit tests cover the CLI parser edge cases beautifully. Ready from my side!

Thanks @Shouryaverma19 , can you please approve, so we can merge 

Image reference:
approve_pr
